### ISSUE TITLE
Error grouping is totally wrong

### ISSUE THREAD

**Issue by livthomas** [bugsnag/bugsnag-js#1284]
### Describe the bug
First of all, I must say we also use Bugsnag in our back-end PHP code and we don't have any problems with it there.
But we see a lot of problems in errors reported to Bugsnag from our front-end code. There are cases when there are multiple unrelated errors grouped into a single Bugsnag issue. But we also see a lot of duplicates, the same errors in multiple Bugsnag issues which could have been merged together. It's nice that Bugsnag tries to be smart and group errors together but it doesn't do very good job in JS code. I guess it's probably caused by the fact that the JS app runs in many different environments (browsers) and so the same error doesn't always look the same.

Coming from Sentry, I would appreciate if I was able to manually group multiple issues together in Bugsnag dashboard and if it was able to learn from these changes. It would also be nice to have an ability to split a single Bugsnag issue into multiple ones in case it contains multiple unrelated errors. If you implement even this, you will be ahead of Sentry. Right now, you're behind.

### Steps to reproduce
1. Track errors in production environment for a while.
2. See many duplicates in Bugsnag dashboard.

### Environment
* Bugsnag version: 7.5.1
* Browser framework version:
    * Vue: 2.6.12
* Browser version (e.g. chrome, safari):any

### Example 

These five errors should have been merged into one:

- https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/5ff7f10c95ef260018c40b81
- https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/5ff7f10c9fe7390018ca199c
- https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/5ff7f10cca9932001962dba1
- https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/5ff7f10cf271250017f720cb
- https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/5ff7f10c9983480018b057f5

**Comment by xander-jones** [comment]
Hey @livthomas, your project's grouping on [surrounding code](https://docs.bugsnag.com/product/error-grouping/#surrounding-code-javascript-only) from where the error is thrown, you can see the code snippet and grouping rationale by hovering over the 'Default Grouping' link on the error in the dash, note we swap out things like variable, and function names for known substitutes to improve grouping when little changes like variable name changes occur.

Whilst the error code and error contexts are the same (with one context anomaly) for these error groups. They all have different error messages, and are getting thrown from different places in your codebase, which is why they're getting grouped into separate errors.

The following errors all have matching time-stamped events. They're 4 separate, but presumably related errors:
* https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/5ff7f10c9fe7390018ca199c
* https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/5ff7f10cca9932001962dba1
* https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/5ff7f10cf271250017f720cb
* https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/5ff7f10c95ef260018c40b81

If these were grouped into one error group, it would likely be much harder to debug, because you could be led to believe that it's just one point in the code that's error'ing, but you can always override the grouping using a few different mechanisms:
* Error Class grouping, e.g. if you wanted all `TypeErrors` to be grouped together: https://docs.bugsnag.com/product/error-grouping/#error-class (note, this method's probably a bit too aggressive to use here as `TypeError` is not unique to just this point in the code, and you'd likely end up over-grouping events, but FYI)
* Error context grouping, e.g. if you wanted all `TypeError` class errors from context `/inventory-update` to be grouped together: https://docs.bugsnag.com/product/error-grouping/#context
* Setting a custom grouping hash: In [a callback](https://docs.bugsnag.com/platforms/javascript/vue/customizing-error-reports/#updating-events-using-callbacks), you can override [the grouping hash](https://docs.bugsnag.com/platforms/javascript/vue/customizing-error-reports/#groupinghash) based on whatever criteria you deem necessary. This is the method that I think would be most suitable in this scenario.

In short, the grouping is working as expected as they're errors originating from different points in your codebase but you may want to consider making some overriding changes to your notifier so that you can force custom grouping where desired.

With regards to manually grouping from the Bugsnag dashboard: this is something on our roadmap, but I don't have an ETA to share at the moment, we'll keep this thread updated as soon as there's news to share 😊

**Comment by livthomas** [comment]
Sorry, that was a bad example. How about these two? Why haven't they been grouped together?

- https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/601afe59025dea0017d412ae
- https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/601afe34a22f8b0017791b28

**Comment by xander-jones** [comment]
Because no source code is available to group by in these errors, we're falling back on a grouping by file, line number, and method: https://docs.bugsnag.com/product/error-grouping/#js-location

* The top of the stack for https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/601afe59025dea0017d412ae is `[native code]:1072:41 parse`
* The top of the stack for https://app.bugsnag.com/snappy-kraken/snappy-app-wp-js/errors/601afe34a22f8b0017791b28 is `[native code]:1030:41 parse`

Because these two errors appear to be coming from different lines in the codebase, they're getting grouped separately.

**Comment by brettrowberry** [comment]
Have you considered https://docs.bugsnag.com/product/error-grouping/#custom-grouping-hash?

**Comment by livthomas** [comment]
@brettrowberry Yes, I started using it after I reported this issue. It is especially useful if you keep track of failed API calls and you don't want them to be randomly merged together into same errors in Bugsnag.

So the situation in our project is much better now. However, I'm still missing an ability to manually merge/split errors in Bugsnag. But since @xander-jones mentioned that it was already on their roadmap, I think we can close this issue.
